core/src/main/java/io/lionweb/serialization/data/MetaPointer.java

```diff
@@ -4,90 +4,84 @@
 import io.lionweb.language.IKeyed;
 import io.lionweb.language.Language;
 import io.lionweb.language.LanguageEntity;
-import java.util.Objects;
+import java.util.*;
 
 /**
  * A MetaPointer is the combination of the pair Language and Version with a Key, which identify one
  * element within that language.
+ *
+ * <p>We should never have multiple instances with the same value, so equality and identity
+ * coincides for this class.
  */
 public class MetaPointer {
-  private String key;
-  private String version;
-  private String language;
+  private static final Map<String, Set<MetaPointer>> INSTANCES_BY_KEY = new HashMap<>();
```

Comment:
This will prevent the garbage collector to claim unused `MetaPointer` instances.

The concept we want here is "interning", Java does it a lot for [Strings](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#intern--)

A quick search turns up articles like https://www.javacodegeeks.com/2014/03/java-object-interning.html that generalizes to arbitrary objects.

Reply:
I have quickly look to interning and also to the flag for deduplicating strings (which helped a bit).
I see the problem with the garbage collection never happening but I am more worried for that in relation to SerializedPropertyValue then for Metapointers.

I will look into interning.

We could also consider keeping only the most recently used values up to a certain size (4096 entries)? But then we would lose the guarantee identity = equality

Reply:
I went for using `WeakHashMap`, so that garbage collection is not prevented by this map